src/connector/src/sink/writer.rs

```diff
@@ -67,6 +67,10 @@ pub trait AsyncTruncateSinkWriter: Send + 'static {
         chunk: StreamChunk,
         add_future: DeliveryFutureManagerAddFuture<'a, Self::DeliveryFuture>,
     ) -> impl Future<Output = Result<()>> + Send + 'a;
+
+    fn barrier(&mut self) -> impl Future<Output = Result<()>> + Send + '_ {
```

Comment:
Better to change the name to `checkpoint`, and only call it when `is_checkpoint=true` for barrier.

Reply:
@Rossil2012 Any reason to flush every ckpt instead of barrier

Reply:
>@Rossil2012 Any reason to flush every ckpt instead of barrier

Flush at checkpoint is compulsory, and flush at barrier is optional.
>Better to change the name to checkpoint, and only call it when is_checkpoint=true for barrier.

I think it's better to leave it `barrier` and pass the `is_checkpoint` as an argument. Align with the semantic of original `SinkWriter` trait.